app/src/ui/diff/diff-explorer.ts

```diff
@@ -1,4 +1,27 @@
-import { DiffLine, DiffHunk } from '../../models/diff'
+import { DiffLine, DiffHunk, DiffLineType } from '../../models/diff'
+
+/**
+ * Indicate the type of changes that are included in the current range.
+ */
+export enum DiffRangeType {
+  /** Only contains added lines. */
+  Additions,
+  /** Only contains added lines. */
+  Deletions,
+  /** Contains both added and removed lines. */
+  Mixed,
+}
+
+/**
+ * Helper object that represents a range of lines in a diff.
+ * Its type represents the type of interactive (added or deleted)
+ * lines that it contains, being null when it has no interactive lines.
+ */
+type DiffRange = {
```

Comment:
Love that you took the time to break this out into its own type. I think conventionally we'd use an interface here instead of a type but I'm absolutely fine with this, it's such an improvement over the anonymous tuple type.